src/qibo/models/dbi/double_bracket.py

```diff
@@ -156,6 +203,98 @@ def hyperopt_step(
         )
         return best["step"]
 
+    def polynomial_step(
+        self,
+        n: int = 4,
+        n_max: int = 5,
+        d: np.array = None,
+        backup_scheduling: DoubleBracketScheduling = None,
```

Comment:
Thanks for the suggestion, the no solution case has been moved to  `choose_step`. However, would you have a good solution for the problem where no additional key arguments can be supplied to the backup scheduling? Meaning that when the backup scheduling runs, only the default values can be used.

Reply:
Maybe, the best approach is to remove the backup scheduling option and handle the failure of the optimization in another way, understanding better why it does not converge. 
For what I have understood, the optimization algorithm fails because the polynomial has not real roots, so I suppose this could mean two things (correct me if I am wrong): either the function has no minima or the approximation is incorrect. 
In the first case you should take one of the range boundaries, in the second one maybe increase the polynomial degree is enough.